python/distrdf/dask/check_friend_trees_alignment.py

```diff
@@ -0,0 +1,77 @@
+import os
+
+import pytest
+
+import ROOT
+from DistRDF.Backends import Dask
+
+
+TREENAMES = [f"distrdf_check_friend_trees_alignment_dask_tree_{i}.root" for i in range(1, 7)]
+FILENAMES = [f"distrdf_check_friend_trees_alignment_dask_file_{i}.root" for i in range(1, 7)]
+
+
+def create_dataset():
+    df = ROOT.RDataFrame(60).Define("x", "rdfentry_")
+
+    ranges = list(range(0, 61, 10))
+    for i in range(len(ranges[:-1])):
+        df.Range(ranges[i], ranges[i+1]).Snapshot(TREENAMES[i], FILENAMES[i])
+
+
+def create_chain():
+
+    main = ROOT.TChain()
+    for i in range(3):
+        main.Add(f"{FILENAMES[i]}?#{TREENAMES[i]}")
+
+    friend = ROOT.TChain()
+    for i in range(3, 6):
+        friend.Add(f"{FILENAMES[i]}?#{TREENAMES[i]}")
+
+    main.AddFriend(friend, "friend")
+
+    return main
+
+
+@pytest.fixture(scope="class")
+def setup_testfriendtreesalignment(request):
+    """
+    Set up test environment for this module. Currently this includes:
+
+    - Write a set of trees usable by the tests in this class
+    - Remove the files at the end of the tests.
+    """
+
+    create_dataset()
+    yield
+    for name in FILENAMES:
+        os.remove(name)
+
+
+@pytest.mark.usefixtures("setup_testfriendtreesalignment")
+class TestDaskFriendTreesAlignment:
+    """Check that distrdf processes the correct entries from friend trees"""
+
+    def test_sum_friend_column(self, connection):
```

Comment:
for my education, where does `connection` come from?

Reply:
`connection` is a pytest fixture, creates a single Dask/Spark connection object for all the tests in the Dask/Spark suite

Reply:
They are defined in the respective folders, for example the Dask one is at https://github.com/root-project/roottest/blob/master/python/distrdf/dask/conftest.py